### PR TITLE
Feat 415 parent operator table

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ZH CARD 415](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/415)

### CODE REVIEW

**Review by marcellmueller — COMMENTED** [review]
Nice work, this is working really well and pretty much good to go aside from that regex warning. 

One thing I noticed while reviewing this that is not caused by this PR is that typing in array fields is extremely slow. I checked the Multiple Operators form to confirm and initially thought it was related to some of my code in the `ArrayFieldTemplate` though after some testing I realized that wasn't the case.

It seems like performance is slow when typing in array fields in general as RJSF re-renders the entire form while typing. That recent PR I did that handles form state externally actually makes this worse. It's not noticeable in fields outside of an array section though since array fields already run kind of slow it becomes extremely noticeable.

Anyways I'm a little split on this and may need to create a tech debt card to handle state externally on every form and update it on submit as you had initially done in the `User Operator` forms.

File: bc_obps/registration/models.py [hunk 1]
```diff
@@ -802,3 +770,61 @@ def save(self, *args, **kwargs):
...
+        db_comment="Index used to differentiate parent operators for the child operator for saving/updating purposes"
+    )
+    legal_name = models.CharField(max_length=1000, db_comment="The legal name of an operator")
+    trade_name = models.CharField(max_length=1000, blank=True, db_comment="The trade name of an operator")
+    cra_business_number = models.IntegerField(db_comment="The CRA business number of an operator")
+    bc_corporate_registry_number = models.CharField(
+        db_comment="The BC corporate registry number of an operator",
+        validators=[
+            RegexValidator(
+                regex="^[A-Za-z]{1,3}\d{7}$",
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply bc_obps/registration/models.py:794]: This is a valid nit. Let me put a quick fix for this.

**Comment by Sepehr-Sobhani** [comment]
> Nice work, this is working really well and pretty much good to go aside from that regex warning.
> 
> One thing I noticed while reviewing this that is not caused by this PR is that typing in array fields is extremely slow. I checked the Multiple Operators form to confirm and initially thought it was related to some of my code in the `ArrayFieldTemplate` though after some testing I realized that wasn't the case.
> 
> It seems like performance is slow when typing in array fields in general as RJSF re-renders the entire form while typing. That recent PR I did that handles form state externally actually makes this worse. It's not noticeable in fields outside of an array section though since array fields already run kind of slow it becomes extremely noticeable.
> 
> Anyways I'm a little split on this and may need to create a tech debt card to handle state externally on every form and update it on submit as you had initially done in the `User Operator` forms.

Yeah, I noticed that yesterday too. Can we add a debounce to the `onchange` function? does it help?

**Comment by marcellmueller** [comment]
> Yeah, I noticed that yesterday too. Can we add a debounce to the `onchange` function? does it help?

https://github.com/bcgov/cas-registration/pull/558

I have this in draft so we handle it on submission similar to how you handled it in the `User Operator` forms instead of onChange. It's still slightly sluggish compared to normal fields though performance is improved a lot from before this update. I think some of it is out of our hands since our array fields are pretty large and complex.

**Review by marcellmueller — APPROVED** [review]
Awesome job on this 🥇

**Pull request merged**